prover/src/gpu/metal/mod.rs

```diff
@@ -401,9 +402,34 @@ impl<
         self.blowup
     }
 
-    /// Returns the trace layout of the execution trace.
-    fn trace_layout(&self) -> &TraceLayout {
-        self.trace_info.layout()
+    /// Populates the provided Lagrange kernel frame starting at the current row (as defined by
+    /// lde_step).
+    /// Note that unlike EvaluationFrame, the Lagrange kernel frame includes only the Lagrange
+    /// kernel column (as opposed to all columns).
+    fn read_lagrange_kernel_frame_into(
+        &self,
+        lde_step: usize,
+        col_idx: usize,
+        frame: &mut LagrangeKernelEvaluationFrame<E>,
+    ) {
+        let frame = frame.frame_mut();
+        frame.truncate(0);
+        let aux_segment = &self.aux_segment_ldes[0];
```

Comment:
We should change `aux_segment_ldes` to be just a single matrix (rather than a vector of matrixes). Same goes for the `aux_segment_trees` field. Basically, these field should look something like this:
```rust
pub struct MetalTraceLde<E: FieldElement<BaseField = Felt>, H: Hasher> {
    ...
    aux_segment_lde: RowMatrix<E>,
    aux_segment_tree: MerkleTree<H>,
    ...
}
```

Reply:
I have set them as optionals